packages/fuel-wallet/src/FuelWalletConnector.ts

```diff
@@ -266,7 +266,12 @@ export class FuelWalletConnector extends FuelConnector {
   }
 
   async currentNetwork(): Promise<Network> {
-    return this.client.request('network', {});
+    const { url } = await this.client.request('network', {});
+    const provider = await Provider.create(url);
+    return {
+      url: provider.url,
+      chainId: provider.getChainId(),
+    };
```

Comment:
fuel wallet should be already returning this

https://github.com/FuelLabs/fuels-wallet/blob/698c272c05ffa99a3554af84db72550dfb97879d/packages/app/src/systems/CRX/background/services/BackgroundService.ts#L341-L352

are you using the latest wallet?

Reply:
[The consumer](https://forum.fuel.network/t/how-to-get-the-network-a-wallet-is-connected-to/6638/5?u=maschad) who reported this bug is not using the wallet but rather the default fuel connector. 

Here's a repro of [the issue](https://github.com/maschad/connector-react-app/blob/main/src/hooks/useWallet.ts#L59-L65)

Reply:
I see, but I think he is using the wallet (as he said in his first comment) but in a older version

It sounds wrong to me because we fixed this issue recently (we’ve also released the select network by chainId) 

I will double check it later using all latest versions (connectors, sdk, wallet)

Reply:
Thanks for confirming @helciofranco